### ISSUE TITLE
MemoryStream dragging memory from previous execution

### ISSUE THREAD

**Issue by xtianjohns** [staltz/xstream#71]
I'm experiencing what seems like odd behavior when working with MemoryStreams and stream completion using xstream.

Consider a stream composed like so:
```js
xs.periodic( 500 ).mapTo( 'world' ).startWith( 'hello' ).take( 2 );
```
I believe that this stream is a MemoryStream due to the `startWith` operator. When I add a listener after this stream has completed, the values appear to be emitted in reverse (`'world'` then `'hello'` instead of `'hello'` then `'world'`).

Here is a [bin](http://www.webpackbin.com/4kWw_TvLb) which demonstrates my experience.

I did some sleuthing and considered that since the MemoryStream holds on to the last emitted value, perhaps it was simply holding on to `'world'` and then emitting the next known value `'hello'`, then emitting a maximum of two of those values and terminating. That behavior would make it appear that the stream had somehow been reversed.

Is that behavior expected? I don't really know how to isolate the question further than that - it may not be an issue with MemoryStream, it may be an issue with take, or it may be not an issue at all.

**Comment by Hypnosphi** [comment]
Seems like memory stream should clear its internal value `_v` on cleanup here: https://github.com/staltz/xstream/blob/master/src/core.ts#L1796-L1799

**Comment by staltz** [comment]
Fixed, released in v5.1.4.

Thanks for the careful report @xtianjohns, and thanks for helping out @Hypnosphi.
